src/test/mocks/archiveLoader.ts

```diff
@@ -85,13 +85,18 @@ export default () => {
       localBooks[`${newBook.id}@${newBook.version}`] = newBook;
       localBookPages[`${newBook.id}@${newBook.version}`] = {};
     },
-    mockPage: (parentBook: ArchiveBook, newPage: ArchivePage, pageSlug: string) => {
+    mockPage: (parentBook: ArchiveBook, newPage: ArchivePage, pageSlug: string, tree?: ArchiveTree) => {
       localBookPages[`${parentBook.id}@${parentBook.version}`][newPage.id] = newPage;
-      localBooks[`${parentBook.id}@${parentBook.version}`].tree.contents.push({
-        id: `${newPage.id}@${newPage.version}`,
-        slug: pageSlug,
-        title: newPage.title,
-      });
+      const currentBook = localBooks[`${parentBook.id}@${parentBook.version}`];
+      if (tree) {
+        currentBook.tree.contents.push(tree);
+      } else {
+        currentBook.tree.contents.push({
+          id: `${newPage.id}@${newPage.version}`,
+          slug: pageSlug,
+          title: newPage.title,
+        });
+      }
```

Comment:
@TomWoodward this way a page is only added directly to a book if no tree is passed in

Reply:
alternatively you could use something like `findArchiveTreeNode` to see if the node already exists before adding it.

i'm not sure how i feel about passing the tree in, since there is nothing that says it would have to be the same tree as the passed book, and if it was different the test page wouldn't work.

Reply:
got it, makes sense